crates/uv/src/commands/pip_compile.rs

```diff
@@ -355,6 +355,26 @@ pub(crate) async fn pip_compile(
                 env::args_os()
                     .skip(1)
                     .map(|arg| arg.normalized_display().to_string())
+                    .scan(None, |skip_next, arg| {
+                        if let Some(true) = skip_next {
+                            *skip_next = None; // reset state and skip this iteration
+                            Some(None)
+                        } else if !include_index_url
+                            && (arg.starts_with("--extra-index-url=")
+                                || arg.starts_with("--index-url="))
+                        {
+                            *skip_next = None; // reset state and skip this iteration
+                            Some(None)
+                        } else if !include_index_url
+                            && (arg == "--extra-index-url" || arg == "--index-url")
+                        {
+                            *skip_next = Some(true); // mark next item to be skipped
+                            Some(None)
+                        } else {
+                            Some(Some(arg)) // keep this argument
+                        }
+                    })
+                    .flatten()
                     .join(" ")
```

Comment:
This looks good to me, though I may pull it out to its own function.

Reply:
Sure, no problem, extracted to a separate function👍